packages/main/src/plugin/index.spec.ts

```diff
@@ -67,18 +72,46 @@ beforeAll(() => {
       },
       app: {
         on: vi.fn(),
+        getVersion: vi.fn(),
       },
       clipboard: {
         writeText: vi.fn(),
       },
+      ipcMain: {
+        handle: vi.fn(),
+        emit: vi.fn().mockReturnValue(true),
+        on: vi.fn(),
+      },
+      BrowserWindow: {
+        getAllWindows: vi.fn(),
+      },
     };
   });
   const trayMenuMock = {} as unknown as TrayMenu;
   pluginSystem = new TestPluginSystem(trayMenuMock, mainWindowDeferred);
 });
 
+const handlers = new Map<string, any>();
+
 beforeEach(() => {
   vi.clearAllMocks();
+  handlers.clear();
+  vi.mocked(ipcMain.handle).mockImplementation((channel: string, listener: any) => {
+    handlers.set(channel, listener);
+  });
+  vi.mocked(BrowserWindow.getAllWindows).mockImplementation(() => {
+    return [
+      {
+        isDestroyed: () => false,
+        webContents,
+      } as unknown as BrowserWindow,
+    ];
+  });
+  vi.mocked(app.getVersion).mockReturnValue('100.0.0');
+  vi.spyOn(Updater.prototype, 'init').mockReturnValue({ dispose: vi.fn() } as any);
```

Comment:
Is any required here ?

Reply:
Turned out there is no need to mock. I just returned an instance of original class.